src/main/scala/firrtl/passes/SplitExpressions.scala

```diff
@@ -27,6 +27,11 @@ object SplitExpressions extends Pass {
     case _ => false
   }
 
+  private def isSignedAdd(e: Expression): Boolean = e match {
+    case DoPrim(PrimOps.Add, _, _, _: SIntType) => true
+    case _ => false
+  }
```

Comment:
This may be prefetching, too far, but maybe we want to do this for any arithmetic op of `SIntType`? E.g., I can produce a similar error with subtraction:

https://www.edaplayground.com/x/WAHv

Reply:
Definitely not prefetching too far. If you can produce the same error with other ops then we should fix those as well. I'm leaning toward the VerilogEmitter change instead since it should fix the whole class of issues but it makes me a little nervous.